tools/src/main/java/org/apache/kafka/tools/CommandOptions.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.kafka.common.utils.Exit;
+
+/**
+ * Base class for command line tools options
+ */
+public abstract class CommandOptions {
+
+    protected final ArgumentParser parser;
+    protected final Namespace ns;
+
+
+    public CommandOptions(String command, String description, String[] args) {
+        this.parser = ArgumentParsers
+                .newArgumentParser(command)
+                .defaultHelp(true)
+                .description(description);
+
+        prepareArgs();
+
+        Namespace ns = null;
+        try {
+            ns = this.parser.parseArgs(args);
+        } catch (ArgumentParserException e) {
+            parser.handleError(e);
+            Exit.exit(1);
```

Comment:
@adyach having the parsing here could be a good idea but exiting from the application maybe not. This should be the base class for other command options classes but changing the application flow could be misleading here (i.e. Exit.exit(1)).

Reply:
@ppatierno I thought about, it is gonna be a basic operation for every command option, if arguments are incorrect for command, I do not see what else you can do except printing the help and exiting, this gonna be repeated in every command option impl. From other hand it is hidden logic, which is not that obvious, but what are alternatives: 1. throw the exception from ctr, which is not that good since every impl will do almost the same, 2. if we really go for the flexibility of parsing options and reaction on the error we could ad another method which is called by default, but can be overridden, like:

protected void handleArgumentParsing(ArgumentParserException e) throws Exception {
    throw new Exception(e);
}
and call in ctr of CommandOption when parsing error happens.
Having CommandOptionsExitOnError class could implement as it is currently in this PR.

Reply:
@adyach I have noticed that this "hidden" logic is what already happens today for the `ConsoleConsumer` tool for example (in the related `ConsumerConfig` class) so I think that we can live with that.
FYI as part of the refactoring for #3453 (for having it more testable), I decided to use that PR for introducing the `CommandOptions` class even getting your good ideas here. In this case when the #3453 will be closed we should have a first version of some of the common components we need for tools refactoring.